webpack/scenes/Subscriptions/UpstreamSubscriptions/UpstreamSubscriptionsPage.js

```diff
@@ -64,10 +81,34 @@ class UpstreamSubscriptionsPage extends Component {
     );
   }
 
+  quantityValidationInput(pool) {
+    if (!pool || pool.updatedQuantity === undefined) return null;
+    if (this.quantityValidation(pool)) {
+      return 'success';
+    }
+    return 'error';
+  }
+
+  validateSelectedRows() {
+    const validated = Array.isArray(this.state.selectedRows) &&
+                      this.state.selectedRows.every(pool => this.quantityValidation(pool));
+
+    if (!validated) {
+      const invalidInputMessage = `Quantities are invalid, please check that the quantities are
+                                   valid numbers above zero and do not exceed available 
+                                   entitlements.`;
+      notify({ message: __(invalidInputMessage), type: 'error' });
+    }
+
+    return validated;
+  }
+
   saveUpstreamSubscriptions() {
+    if (!this.validateSelectedRows()) return;
+
     const updatedPools = _.map(
       this.state.selectedRows,
-      pool => ({ ...pool, quantity: pool.updatedQuantity }),
+      pool => ({ ...pool, quantity: parseInt(pool.updatedQuantity, 10) }),
```

Comment:
I didn't even think to try a hex or octal number, but looks like you already accounted for such mischief

Reply:
I tried to add "011" and it deducted 11 as expected